src/chunk_append/exec.c

```diff
@@ -449,7 +449,15 @@ choose_next_subplan_non_parallel(ChunkAppendState *state)
 static void
 choose_next_subplan_for_leader(ChunkAppendState *state)
 {
-	state->current = NO_MATCHING_SUBPLANS;
+	/*
+	 * If no workers got launched for this parallel plan
+	 * we have to let leader participate in subplan
+	 * execution.
+	 */
+	if (state->pcxt->nworkers_launched == 0)
+		choose_next_subplan_for_worker(state);
```

Comment:
Is this new behaviour tested? How is it difficult to add new test?

Reply:
It would require quite a bit of extra infrastructure to produce a test case.

Reply:
Thank you for clarifying.